Checkout/Sample/Source/Card Tokenisation/Card Details/CardDetailsTableViewCell.swift

```diff
@@ -7,30 +7,129 @@
 
 import UIKit
 
+protocol CardDetailsTableViewCellDelegate: AnyObject {
+  var cardNumberDelegate: UITextFieldDelegate { get }
+}
+
 class CardDetailsTableViewCell: UITableViewCell {
   @IBOutlet weak var cardNumberTextField: UITextField!
   @IBOutlet weak var expiryMonthTextField: UITextField!
   @IBOutlet weak var expiryYearTextField: UITextField!
   @IBOutlet weak var cvvTextField: UITextField!
+  @IBOutlet weak var expiryTextFieldStackView: UIStackView!
+  @IBOutlet weak var expiryDatePickerField: UITextField!
+
+  struct ViewModel {
+    var onCardNumberChange: ((String?) -> Void)?
+    var onCVVChange: ((String?) -> Void)?
+
+    var onExpiryMonthStringChange: ((String?) -> Void)?
+    var onExpiryYearStringChange: ((String?) -> Void)?
+
+    var onExpiryIntChange: ((_ month: Int, _ year: Int) -> Void)?
+  }
+
+  var viewModel: ViewModel? {
+    didSet {
+      expiryDatePickerField.inputView = expirationDatePicker
+      expiryDatePickerField.inputAccessoryView = datePickerToolbar
+      expiryDatePickerField.delegate = noTextEntryUITextFieldHelper
+    }
+  }
+
+  private let expirationDatePicker = ExpirationDatePicker()
+  private let noTextEntryUITextFieldHelper = NoTextEntryUITextFieldHelper()
+  private let dateFormatter = DateFormatter()
 
-  var onCardNumberChange: ((String?) -> Void)?
-  var onExpiryMonthChange: ((String?) -> Void)?
-  var onExpiryYearChange: ((String?) -> Void)?
-  var onCVVChange: ((String?) -> Void)?
+  private lazy var datePickerToolbar: UIToolbar = {
+    let pickerToolbar = UIToolbar()
+    pickerToolbar.autoresizingMask = .flexibleHeight
+
+    let flexSpace = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+    let doneButton = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(Self.dateEditingDone))
+
+    // add the items to the toolbar
+    pickerToolbar.items = [flexSpace, doneButton]
+
+    return pickerToolbar
+  }()
+
+  required init?(coder: NSCoder) {
+    super.init(coder: coder)
+
+    expirationDatePicker.viewModel = ExpirationDatePicker.ViewModel(onDateChanged: onDateChanged)
+    dateFormatter.dateFormat = "MM/YYYY"
+  }
+
+  private func onDateChanged(_ month: Int, _ year: Int) {
+    guard let date = date(month: month, year: year) else {
+      return
+    }
+
+    expiryDatePickerField.text = dateFormatter.string(from: date)
+    viewModel?.onExpiryIntChange?(month, year)
+  }
+
+  private func date(month: Int, year: Int) -> Date? {
+    var dateComponents = DateComponents()
+
+    dateComponents.month = month
+    dateComponents.year = year
+    dateComponents.calendar = Calendar(identifier: .gregorian)
+    dateComponents.timeZone = .utc
+
+    return dateComponents.date
+  }
+
+  weak var delegate: CardDetailsTableViewCellDelegate? {
+    didSet {
+      guard let delegate = delegate else {
+        return
+      }
+
+      cardNumberTextField.delegate = delegate.cardNumberDelegate
```

Comment:
It would be nice to keep code a little organised and move this property around other properties, potentially grouping by access control for quick discovery.

Also, if delegate is set to nil, shouldn't that also be passed to the `cardNumberTextField`? I'd think the guard is not required.

But good use of the property observer, I'd say this is the best way to use it, to pass a state change along to other dependencies, not to trigger logic flows.